detectable/src/main/java/com/synopsys/integration/detectable/detectables/pipenv/parse/PipfileLockExtractor.java

```diff
@@ -0,0 +1,41 @@
+package com.synopsys.integration.detectable.detectables.pipenv.parse;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+
+import com.google.gson.Gson;
+import com.synopsys.integration.bdio.graph.DependencyGraph;
+import com.synopsys.integration.detectable.detectable.codelocation.CodeLocation;
+import com.synopsys.integration.detectable.detectables.pipenv.parse.model.PipfileLock;
+import com.synopsys.integration.detectable.detectables.pipenv.parse.model.PipfileLockDependency;
+import com.synopsys.integration.detectable.extraction.Extraction;
+
+public class PipfileLockExtractor {
+    private final Gson gson;
+    private final PipfileLockParser pipfileLockParser;
+    private final PipfileLockDependencyTransformer pipfileLockTransformer;
+
+    public PipfileLockExtractor(
+        Gson gson,
+        PipfileLockParser pipfileLockParser,
+        PipfileLockDependencyTransformer pipfileLockTransformer
+    ) {
+        this.gson = gson;
+        this.pipfileLockParser = pipfileLockParser;
+        this.pipfileLockTransformer = pipfileLockTransformer;
+    }
+
+    public Extraction extract(File pipfileLockFile, PipfileLockDetectableOptions pipfileLockDetectableOptions) throws IOException {
```

Comment:
Detectable shouldn't know about Options files. The Detectable classes should always take in just what they need via constructors.